### PR TITLE
#266 A user should be able to comment and react to a blog

### PR DESCRIPTION
**Pull request by Philimuhire**
### PR Description
A user should be able to comment and react to blog posts.

### Description of tasks that were expected to be completed
This PR introduces functionality allowing users to:
- Add comments to a blog post.
- Fetch all comments related to a given blog post.
- React to blog posts with reactions (e.g., Like, Love, Support, Funny).
- Like individual comments.
- Add replies to specific comments.

### Related Issues
https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/266

### Changes Made:
- Added a model to handle reactions at the database level.
- Created a schema to define the GraphQL schema for comments.
- Created src/schema/reactionSchema.ts to define the GraphQL schema for reactions.
- Added a resolver for creating comments.
- Added a resolver for managing likes on comments.
- Added a resolver for for managing reactions on both blog posts.

### How to test this PR
- Clone the repository
- Run npm install to install dependencies and navigate to the `ft-comments-reactions-bn` branch
- Start the development server using npm run dev.
- Perform any reaction or comment query or mutation and verify the changes reflect in the database.

### Screenshoots
image
image
image
image

### CODE REVIEW

**Review by senseidudu — APPROVED** [review]

**Review by Ishimwe7 — APPROVED** [review]
good job @Philimuhire

File: src/resolvers/blogResolvers.ts
Comment (MnorbertVii) [inline src/resolvers/blogResolvers.ts:1]: and so why did you remove these lines of codes to hide a blog?

File: src/schema/blogSchema.ts
Comment (MnorbertVii) [inline src/schema/blogSchema.ts:1]: why did you remove the hideBlog mutation line 83 and why did you add isHidden field on line 91?

**Review by MnorbertVii — CHANGES_REQUESTED** [review]

File: src/resolvers/blogResolvers.ts
Comment (Philimuhire) [reply src/resolvers/blogResolvers.ts:1]: resolved, kindly check

File: src/schema/blogSchema.ts
Comment (Philimuhire) [reply src/schema/blogSchema.ts:1]: resolved, kindly check

**Review by MnorbertVii — APPROVED** [review]

**Review by Mugisha146 — APPROVED** [review]

**Review by SaddockAime — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]
